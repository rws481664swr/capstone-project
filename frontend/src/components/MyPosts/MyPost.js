import Timestamp from "../General/Timestamp";
import {useState} from "react";
import DisplayProfile from "../../pages/user/UserProfile/DisplayProfile";

const MyPost = ({post,onClick}) => {

    return(
        <>

    <div onClick={onClick}
         className={'MyPost'} key={post._id}>
        <div className={'MyPostTitle'}>{post.title}</div>
        <div className={'MyPostContent'}>
            {/*{post.content}*/}
           Posted <Timestamp className={'MyPostTimestamp'} date={post.postDate}/>
        </div>
    </div>
        </>
)}
export default MyPost