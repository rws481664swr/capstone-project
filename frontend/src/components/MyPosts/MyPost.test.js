import MyPost from "./MyPost";
import { render, screen } from "@testing-library/react";

describe("MyPost", () => {
    it('should render', function () {
        const {getByText}=render(<MyPost
            onClick={jest.fn()}
            post={
            {
                "title": "My Post",
                "course": "5f9d5c6b2f1c1e1b1c1f1d1e",
                "user": "5f9d5c6b2f1c1e1b1c1f1d1e",
                "content": "This is my post",
                "pinned": false,
                "postDate": "2020-10-30T18:25:23.000Z",
                "_id": "5f9d5c6b2f1c1e1b1c1f1d1e"
            }
        }/>);
        expect(getByText('My Post')).toBeInTheDocument();
        // expect(getByText( "This is my post")).toBeInTheDocument();
    });
    it('should render', function () {
        const {asFragment}=render(
            <MyPost
                onClick={jest.fn()} post={
            {
                "title": "My Post",
                "course": "5f9d5c6b2f1c1e1b1c1f1d1e",
                "user": "5f9d5c6b2f1c1e1b1c1f1d1e",
                "content": "This is my post",
                "pinned": false,
                "postDate": "2020-10-30T18:25:23.000Z",
                "_id": "5f9d5c6b2f1c1e1b1c1f1d1e"
            }
        }/>
        );
        expect(asFragment()).toMatchSnapshot()

    });
})
