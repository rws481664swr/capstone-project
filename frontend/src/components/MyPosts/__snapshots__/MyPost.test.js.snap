// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPost should render 1`] = `
<DocumentFragment>
  <div
    class="MyPost"
  >
    <div
      class="MyPostTitle"
    >
      My Post
    </div>
    <div
      class="MyPostContent"
    >
      Posted 
      <span
        class="MyPostTimestamp"
      >
          at  2:25 PM on 10/30
      </span>
    </div>
  </div>
</DocumentFragment>
`;
